source/tests/test.ts

```diff
@@ -739,40 +739,73 @@ test('is.odd', t => {
 	}
 });
 
-test('is.empty', t => {
-	t.true(m.empty(null));
-	t.true(m.empty(undefined));
+test('is.emptyArray', t => {
+	t.true(m.emptyArray([]));
```

Comment:
Why not use testType for all these examples? The whole point of the change is to make sure that the methods work on their respective type only. This doesn't do that. Refers to all the empty/nonEmpty tests

Reply:
@gioragutt

Reply:
@ArfatSalman

You can see that in the `is.plainObject` tests:
```typescript
test('is.plainObject', t => {
	testType(t, 'plainObject', ['object', 'promise']);
});
```

The `object` and `promise` tests are excluded.
What `testType` does, it run the the test you specified in the [`types`](https://github.com/sindresorhus/is/blob/master/source/tests/test.ts#L28) array, and check that the given predicate works for the fixtures your provided, but not for all the others.

In `plainObject`'s case, there are tests for different methods that succeed (`object` and `promise`), thus they are excluded.

You should add `is.<yourMethod>` to the `types` array.
I would reconsider doing it to `nonEmpty` methods, since they might have a little too much exceptions.
Thus, for the `emptyX` tests, use the `testType` method and add your cases to the `types` array,
and for the `nonEmpty` you should write specific assertions.

@sindresorhus sounds good?

Reply:
@gioragutt 👍